mongo/bulk_write.go

```diff
@@ -248,7 +248,7 @@ func (bw *bulkWrite) runDelete(ctx context.Context, batch bulkWriteBatch) (opera
 func createDeleteDoc(filter interface{}, collation *options.Collation, hint interface{}, deleteOne bool,
 	registry *bsoncodec.Registry) (bsoncore.Document, error) {
 
-	f, err := transformBsoncoreDocument(registry, filter)
+	f, err := transformBsoncoreDocument(registry, filter, true, "")
```

Comment:
Let's pass the parameter names down everywhere. We shouldn't be making assumptions about what `transformBsoncoreDocument` is doing internally, and it'll make it easier when we make the changes I proposed earlier to `ErrNilDocument`.

Reply:
done